src/modules/ekf2/EKF2.hpp

```diff
@@ -264,6 +264,8 @@ class EKF2 final : public ModuleParams, public px4::ScheduledWorkItem
 	uORB::SubscriptionCallbackWorkItem _sensor_combined_sub{this, ORB_ID(sensor_combined)};
 	uORB::SubscriptionCallbackWorkItem _vehicle_imu_sub{this, ORB_ID(vehicle_imu)};
 
+	uORB::SubscriptionMultiArray<distance_sensor_s> _distance_sensor_subs{ORB_ID::distance_sensor};
```

Comment:
Delete the existing `_distance_sensor_sub` and used this instead.

Reply:
> Delete the existing `_distance_sensor_sub` and used this instead.

I was considering this; I just didn't quite understand all the parts here. What is the purpose of this if, is there some situation where the selection can fail?
						int ndist = orb_group_count(ORB_ID(distance_sensor));

						if (ndist > 1) {
							PX4_INFO("%d - selected distance_sensor:%d (%d advertised)", _instance, i, ndist);
						}

Reply:
just added a simple deletion of _distance_sensor_sub on the topmost commit, still need to test as well

Reply:
Tested also the removal of _distance_sensor_sub, and observed no issues with height, but I had only one distance sensor HW at hand.